---
title: C6318 | Microsoft Docs
ms.date: 11/15/2016
ms.prod: visual-studio-dev14
ms.technology: vs-ide-code-analysis
ms.topic: reference
f1_keywords:
- C6318
helpviewer_keywords:
- C6318
ms.assetid: 3284a83e-bb8e-461c-adcc-cfc66ceea05e
caps.latest.revision: 15
author: mikeblome
ms.author: mblome
manager: jillfra
ms.openlocfilehash: fcf3cd31b998a235b302b265228c89fbdbb64a41
ms.sourcegitcommit: 8b538eea125241e9d6d8b7297b72a66faa9a4a47
ms.translationtype: MT
ms.contentlocale: zh-TW
ms.lasthandoff: 01/23/2019
ms.locfileid: "58944485"
---
# <a name="c6318"></a>C6318
[!INCLUDE[vs2017banner](../includes/vs2017banner.md)]

警告 C6318:定義錯誤的 __try /\__except： 使用常數 EXCEPTION_CONTINUE_SEARCH 或評估為零例外狀況篩選條件運算式中的另一個常數。 例外狀況處理常式區塊中的程式碼不會執行  
  
 這則警告指出，此結構化例外狀況處理常式的受保護區塊中發生的例外狀況，例外狀況並不會處理因為常數`EXCECPTION_CONTINUE_SEARCH`例外狀況篩選條件運算式中使用。  
  
 此程式碼就相當於受保護區塊，而不需要例外狀況處理常式區塊的因為處理常式區塊不會執行。  
  
## <a name="example"></a>範例  
 下列程式碼會產生這個警告：  
  
```  
#include <excpt.h>  
#include <stdio.h>  
  
void f (char *pch)  
{  
  __try   
  {  
    // assignment might fail  
    *pch = 0;  
  }   
  __except (EXCEPTION_CONTINUE_SEARCH) // warning 6318  
  {  
    puts("Exception Occurred");  
  }  
}  
```  
  
 若要更正這個警告，請使用下列程式碼：  
  
```  
#include <excpt.h>  
#include <stdio.h>  
#include <windows.h>  
  
void f (char *pch)  
{  
  __try   
  {  
    // assignment might fail  
    *pch = 0;  
  }   
  __except( (GetExceptionCode() == EXCEPTION_ACCESS_VIOLATION) ?  
                EXCEPTION_EXECUTE_HANDLER : EXCEPTION_CONTINUE_SEARCH )  
  {  
    puts("Access violation");  
  }  
}   
```  
  
## <a name="see-also"></a>另請參閱  
 [try-except 陳述式](http://msdn.microsoft.com/library/30d60071-ea49-4bfb-a8e6-7a420de66381)
